Trigger: Follow up when order note is added [STOMAIL-7366]

pavel-mailpoet opened the pull request:
## Description

When an order note gets added, the trigger is triggered.

<img width="2262" height="1220" alt="Screenshot 2025-07-16 at 13 59 33" src="https://github.com/user-attachments/assets/3c45e56e-c7ea-41e9-ab1a-6d2bc8e7839e" />


## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

Closes:  [STOMAIL-7366]

## After-merge notes

_N/A_

## Tasks

- [x] I have added a changelog entry (`./do changelog:add --type=<type> --description=<description>`)
- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:stomail-7366-trigger-follow-up-when-order-note-is-added)

_The latest successful build from `stomail-7366-trigger-follow-up-when-order-note-is-added` will be used. If none is available, the link won't work._

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

* **New Features**
  * Introduced a new WooCommerce automation trigger that activates when an order note is added, with filters for note type and content.
  * Added a user-friendly interface to configure the trigger’s settings, including note type selection and content matching.

* **Documentation**
  * Enhanced documentation for automation step and trigger interfaces to clarify their purpose and usage.

* **Tests**
  * Added thorough integration tests validating the new order note trigger’s behavior under various conditions.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

pavel-mailpoet committed:
Add a new trigger for order note created

[STOMAIL-7366]

pavel-mailpoet committed:
Add Order Note settings

[STOMAIL-7366]

pavel-mailpoet committed:
Add an integration test

[STOMAIL-7366]

pavel-mailpoet committed:
Add docsblock for Trigger interface

[STOMAIL-7366]

pavel-mailpoet committed:
Add docsblock for Step interface

[STOMAIL-7366]

pavel-mailpoet committed:
Add changelog

[STOMAIL-7366]

pavel-mailpoet committed:
Improve the namespace for the test

pavel-mailpoet committed:
Add a validation

NeosinneR committed:
Unify trigger title case with other triggers

[STOMAIL-7366]

NeosinneR commented on mailpoet/lib/Automation/Integrations/WooCommerce/Triggers/Orders/OrderNoteAddedTrigger.php:
```diff
@@ -0,0 +1,137 @@
+    if ($configuredNoteType !== 'all') {
+      // phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+      $isCustomerNote = (bool)get_comment_meta((int)$comment->comment_ID, 'is_customer_note', true);
+      $actualNoteType = $isCustomerNote ? 'customer' : 'private';
```
This is an unnecessary database call (as well as duplicated code) as the data should already be present in the `CommentSubject`.

NeosinneR reviewed (commented):
Looks and functions as expected. I also tested trigger filters to further filter the trigger by order data. I fixed title case in https://github.com/mailpoet/mailpoet/pull/6262/commits/266f1ccee55bc52bac5bed700a206263e74e596d but notice one other thing that could use refactor (comment below).
 
<img width="645" height="501" alt="Screenshot 2025-07-21 at 14 34 03" src="https://github.com/user-attachments/assets/9edd243f-f68b-4463-87cb-90790ed06e74" />

pavel-mailpoet committed:
Use note_type from subject args in OrderNoteAddedTrigger

Avoids unnecessary database calls by retrieving note_type from the CommentSubject’s args.

[STOMAIL-7366]

Aschepikov reviewed (approved):
GJ!
